cashu/mint/app.py

```diff
@@ -29,20 +29,27 @@
 
 @asynccontextmanager
 async def lifespan(_: FastAPI) -> AsyncIterator[None]:
-    await start_mint()
-    if settings.mint_require_auth:
-        await start_auth()
     try:
-        yield
-    except asyncio.CancelledError:
-        # Handle the cancellation gracefully
-        logger.info("Shutdown process interrupted by CancelledError")
-    finally:
+        await start_mint()
+        if settings.mint_require_auth:
+            await start_auth()
         try:
-            await redis.disconnect()
-            await shutdown_mint()
+            yield
         except asyncio.CancelledError:
-            logger.info("CancelledError during shutdown, shutting down forcefully")
+            # Handle the cancellation gracefully
+            logger.info("Shutdown process interrupted by CancelledError")
+        finally:
+            try:
+                await redis.disconnect()
+                await shutdown_mint()
+            except asyncio.CancelledError:
+                logger.info("CancelledError during shutdown, shutting down forcefully")
+    except BackendConnectionError as e:
+        logger.error(f"Failed to start mint: {e}")
+        # We still need to yield to allow FastAPI to continue
+        # but the app will return errors for all endpoints
+        yield
+        # We don't need to shut down anything as startup failed
```

Comment:
Do we need to try: except: the entire code block or would it be enough to do it only with `await start_mint()`?